index.bs

```diff
@@ -0,0 +1,221 @@
+
+
+
+
+{{LayoutShift}} interface {#sec-layout-shift}
+=======================================
+
+<pre class="idl">
+    interface LayoutShift : PerformanceEntry {
+      readonly attribute long value;
+    };
+</pre>
+
+{{LayoutShift}} extends the following attributes of {{PerformanceEntry}} interface:
+
+* The {{PerformanceEntry/name}} attribute must return the {{DOMString}} <code>"layout-shift"</code>
+* The {{PerformanceEntry/entryType}} attribute must return <code>"layoutShift"</code>.
+* The {{PerformanceEntry/startTime}} attribute must return a {{DOMHighResTimeStamp}} of when the shift occured.
+* The {{PerformanceEntry/duration}} attribute must return 0.
+* The <dfn attribute for=LayoutShift>value</dfn> attribute's getter will return the value it is set to.
+
+<dfn attribute for=PerformanceLongTaskTiming>attribution</dfn>
+A user agent implementing {{LayoutShift}} must perform the following steps:
+1. Run the <a>register a performance entry type</a> algorithm with <code>"layoustShift"</code> as input.
+
+
+NOTE: The <a>getEntriesByType</a> method will not return entries of this type.  To obtain instability events that occur before a <a>PerformanceObserver</a> has been registered, register the observer with <a>buffered</a> set to true.
+
+
+
+
+Processing model {#sec-processing-model}
+========================================
+
+
+Modifications to the HTML specification {#sec-modifications-HTML}
+--------------------------------------------------------
+
+<em>This section will be removed once the <a href=https://html.spec.whatwg.org/multipage>HTML specification</a> has been modified.</em>
+
+For each <a>element</a>, there is a:
+* <dfn>currentFrameStartingPoint</dfn> which is initially set to <code>null</code>.
+* <dfn>previousFrameStartingPoint</dfn> which is initially set to <code>null</code>.
+* <dfn>currentVisualRepresentation</dfn> which is initially set to <code>null</code>.
+* <dfn>previousVisualRepresentation</dfn> with is initially set to <code>null</code>.
+
+<div algorithm="additions to update rendering">
+    In the <a>update the rendering</a> step of the <a>event loop processing model</a>, add a step right after the step that calls <a>mark paint timing</a>:
+
+    1. Invoke the algorithm to <a>evaluate the layout instability</a> of a document with the complete set of <a>element</a>s for that {{Document}}.
+</div>
+
+
+Evaluate Layout Instability {#sec-eval-layout-instability}
+--------------------------------------------------------
+
+<div algorithm="evaluate the layout instability">
+    When asked to <dfn export>evaluate the layout instability</dfn>, with <var>elements</var> as the list of elements to be evaluated, run the following steps:
+
+    1. Let <var>unstable elements</var> be an empty list.
+    1. Let <var>total width</var> be the current <a>viewport</a> width.
```

Comment:
This is underdefined. We need a link for the 'viewport width'. It's not clear to me what the coordinates would be. Ditto for height below.

Reply:
There is a link for viewport.  I am not sure What a link for the whole width/height would look like. 

@skobes Adding steve for input

Reply:
https://www.w3.org/TR/CSS21/visuren.html#viewport

The units should match the way we measure the visual rects of elements (CSS pixels?)

Reply:
What I mean is that the link does not tell me how to compute the width or height of the viewport, or where it comes from. Maybe this is fine? I'm not sure.

Reply:
Given the [amount of discussion](https://github.com/w3c/IntersectionObserver/issues/95) that's taken place in the IntersectionObserver issues over the visual viewport vs. the layout viewport. I think it's best if we're specific.

Reply:
@skobes How is this computed in the code?

Reply:
The implementation is currently using the visual viewport.

Unfortunately the layout vs visual viewport concepts aren't currently spec'ed.  I asked bokan@ and he said the closest thing we have is some non-normative descriptions in https://wicg.github.io/visual-viewport/index.html

Reply:
Ok I added in links to the visual viewport width/height which is the closest that we have at the moment.